Tools/TestWebKitAPI/Tests/WebKit/WKPageHasMediaStreamingActivity.mm

```diff
@@ -0,0 +1,137 @@
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+
+#if PLATFORM(COCOA)
+
+#import "PlatformUtilities.h"
+#import "PlatformWebView.h"
+#import "Test.h"
+#import <JavaScriptCore/JavaScriptCore.h>
+#import <WebKit/WKPagePrivate.h>
+#import <WebKit/WKPreferencesRef.h>
+#import <WebKit/WKPreferencesRefPrivate.h>
+#import <WebKit/WKSerializedScriptValue.h>
+
+// This test loads file-with-video.html. Then it calls a JavaScript method to create a source buffer and play the video,
+// waits for WKMediaNetworkingActivity notification to be fired.
+
+namespace TestWebKitAPI {
+
+#if WK_HAVE_C_SPI
+
+static NSString * const WebKitMediaStreamingActivity = @"WebKitMediaStreamingActivity";
+
+static bool isMSEEnabledChanged;
+static bool isMSEEnabled;
+static bool didFinishLoad;
+
+static void nullJavaScriptCallback(WKSerializedScriptValueRef, WKErrorRef error, void*)
+{
+}
+
+static void isMSEEnabledCallback(WKSerializedScriptValueRef serializedResultValue, WKErrorRef error, void*)
+{
+    JSGlobalContextRef scriptContext = JSGlobalContextCreate(0);
+
+    JSValueRef resultValue = WKSerializedScriptValueDeserialize(serializedResultValue, scriptContext, 0);
+    EXPECT_TRUE(JSValueIsBoolean(scriptContext, resultValue));
+
+    isMSEEnabledChanged = true;
+    isMSEEnabled = JSValueToBoolean(scriptContext, resultValue);
+
+    JSGlobalContextRelease(scriptContext);
+}
+
+static void didFinishNavigation(WKPageRef page, WKNavigationRef, WKTypeRef, const void*)
+{
+    didFinishLoad = true;
+}
+
+static void setUpClients(WKPageRef page)
+{
+    WKPageNavigationClientV0 loaderClient;
+    memset(&loaderClient, 0, sizeof(loaderClient));
+
+    loaderClient.base.version = 0;
+    loaderClient.didFinishNavigation = didFinishNavigation;
+
+    WKPageSetPageNavigationClient(page, &loaderClient.base);
+}
+
+TEST(WebKit, MSEHasMediaNetworkingActivity)
```

Comment:
This test is using WK API while this is a .mm file.
If we are in a .mm file, it would make more sense to use a regular ObjC API, like `[webView loadTestPageNamed]` or `[webView stringByEvaluatingJavaScript:@"playVideo()"];`

Reply:
done